### PR TITLE
reference: add documentation for statement summary

### PR DESCRIPTION
**Pull request by djshow832**
### What is changed, added or deleted? 
Add documentation for statement summary.

Statement summary is recently released, and a documentation is needed to guide users.

### What is the related PR or file link(s)? 
master: pingcap/tidb#12017, pingcap/tidb#12217
release 3.0: pingcap/tidb#12306, pingcap/tidb#12308
release 3.1: pingcap/tidb#12328, pingcap/tidb#12499

### Which version does your change affect? 
dev, v3.0

### CODE REVIEW

File: dev/reference/performance/statement-summary.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+statement summary 功能默认关闭，通过设置系统变量打开。例如：
+
+```sql
+set global tidb_enable_stmt_summary = true;
+```
+
+`tidb_enable_stmt_summary` 有 global 和 session 两种作用域，它们的生效方式与其他系统变量不一样：
+
+- 设置 global 变量后整个集群立即生效
+- 设置 session 变量后当前节点立即生效，这对于调试单个节点比较有用
```
Comment (bb7133) [inline dev/reference/performance/statement-summary.md]:
```suggestion
- 设置 session 变量后当前 TiDB-Server 立即生效，这对于调试单个 TiDB-Server 比较有用
```
我觉得这样会更准确一些

File: v3.0/reference/configuration/tidb-server/configuration-file.md [hunk 2]
```diff
@@ -386,3 +386,17 @@ TiDB 服务状态相关配置。
 
 + 输与 database 相关的 QPS metrics 到 promethus的开关。
 + 默认值：false
+
+## stmt-summary
```
Comment (bb7133) [inline v3.0/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
## stmt-summary  <span class="version-mark">从 v3.0.4 版本开始引入</span>
```

File: dev/reference/configuration/tidb-server/configuration-file.md [hunk 3]
```diff
@@ -385,3 +385,17 @@ TiDB 服务状态相关配置。
 
 + 输与 database 相关的 QPS metrics 到 promethus的开关。
 + 默认值：false
+
+## stmt-summary
```
Comment (bb7133) [inline dev/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
## stmt-summary  <span class="version-mark">从 v3.0.4 版本开始引入</span>
```

**Comment by bb7133** [comment]
Hi @lilin90 @yikeke, please help to review this PR, thanks!

File: dev/reference/configuration/tidb-server/configuration-file.md (hunk 3 above)
Comment (djshow832) [reply dev/reference/configuration/tidb-server/configuration-file.md]: Addressed.

File: v3.0/reference/configuration/tidb-server/configuration-file.md (hunk 2 above)
Comment (djshow832) [reply v3.0/reference/configuration/tidb-server/configuration-file.md]: Addressed.

File: dev/reference/performance/statement-summary.md (hunk 1 above)
Comment (djshow832) [reply dev/reference/performance/statement-summary.md]: Addressed.

**Comment by lilin90** [comment]
@dcalvin Please help review this PR. Thanks!

File: dev/reference/performance/statement-summary.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
+---
+title: Statement Summary Table
+category: reference
+---
+
+# Statement Summary Table
+
+## 前言
```
Comment (dcalvin) [inline dev/reference/performance/statement-summary.md]:
```suggestion
## 简介
```
“前言” 这样的词不太符合技术文档的风格

File: dev/reference/performance/statement-summary.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+我们在排查线上 TiDB 问题时，可能有下面这样的需求：
+
+- SQL 延迟比较大，是不是服务端的问题？
+- 哪类 SQL 的总耗时最高？
+
+遇到这类需要定位 SQL 的问题，我们会首先想到两种方式来排查：
+
+- 打开 general log，但是打印 general log 对 Server 的性能会有影响。
+- `admin show slow` 显示慢查询日志，但是有问题的 SQL 可能没有被归为慢 SQL。
+
```
Comment (dcalvin) [inline dev/reference/performance/statement-summary.md]: Suggest we remove this kind of info, which is redundant to users. In this section, we only need to introduce what is statement summary table and what this document is about. @djshow832

File: dev/reference/performance/statement-summary.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+针对于这些 SQL 类的性能问题，MySQL 在 `performance_schema` 提供了 [statement summary tables](https://dev.mysql.com/doc/refman/5.6/en/statement-summary-tables.html)，用来监控和统计 SQL。例如其中的一张表 `events_statements_summary_by_digest`，提供了丰富的字段，包括延迟、执行次数、扫描行数、全表扫描次数等，有助于用户定位 SQL 问题。
```
Comment (dcalvin) [inline dev/reference/performance/statement-summary.md]:
```suggestion
针对 SQL 性能相关的问题，MySQL 在 `performance_schema` 提供了 [statement summary tables](https://dev.mysql.com/doc/refman/5.6/en/statement-summary-tables.html)，用来监控和统计 SQL。例如其中的一张表 `events_statements_summary_by_digest`，提供了丰富的字段，包括延迟、执行次数、扫描行数、全表扫描次数等，有助于用户定位 SQL 问题。
```

File: dev/reference/performance/statement-summary.md [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+|     7855660 |     1122237 |          7 | select avg(salary) from employee where company_id=2013                |
+|     7241960 |     1448392 |          5 | select * from employee join company on employee.company_id=company.id |
+|     2084081 |     1042040 |          2 | select * from employee where name='eric'                              |
++-------------+-------------+------------+-----------------------------------------------------------------------+
+3 rows in set (0.00 sec)
+```
+
+## 参数配置
+
+statement summary 功能默认关闭，通过设置系统变量打开。例如：
```
Comment (dcalvin) [inline dev/reference/performance/statement-summary.md]:
```suggestion
statement summary 功能默认关闭，通过设置系统变量打开，例如：
```

File: dev/reference/performance/statement-summary.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+由于 `events_statements_summary_by_digest` 是内存表，为了防止内存问题，需要限制保存的 SQL 条数和 SQL 的最大显示长度。这两个参数都在 config.toml 的 [stmt-summary] 类别下配置：
+
+- 通过 `max-stmt-count` 更改保存的 SQL 种类数量，默认 100 条。当 SQL 种类超过 `max-stmt-count` 时，会移除最近没有使用的 SQL。
+- 通过 `max-sql-length` 更改 `DIGEST_TEXT` 和 `QUERY_SAMPLE_TEXT` 的最大显示长度，默认是 4096。
+
+这两个参数建议根据实际情况调整，不宜设置得过大。
+
+## 目前的限制
+
+`events_statements_summary_by_digest` 现在还存在一起限制：
```
Comment (dcalvin) [inline dev/reference/performance/statement-summary.md]:
```suggestion
`events_statements_summary_by_digest` 现在还存在一些限制：
```

File: dev/reference/performance/statement-summary.md [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+| MIN_LATENCY       | 这类 SQL 执行的最小延迟，单位 ns |
+| AVG_LATENCY       | 这类 SQL 执行的平均延迟，单位 ns |
+| SUM_ROWS_AFFECTED | 这类 SQL 的总影响行数    |
+| FIRST_SEEN        | 这类 SQL 第一次执行的时间 |
+| LAST_SEEN         | 这类 SQL 最后一次执行的时间 |
+| QUERY_SAMPLE_TEXT | 这类 SQL 首次出现的原 SQL 语句 |
+
+## 排查示例
+
+对于文章开头描述的几个问题，下面来演示如何利用 statement summary 来排查。
```
Comment (dcalvin) [inline dev/reference/performance/statement-summary.md]:
```suggestion
下面来两个示例问题演示如何利用 statement summary 来排查。
```

File: dev/reference/performance/statement-summary.md (hunk 5 above)
Comment (dcalvin) [reply dev/reference/performance/statement-summary.md]: This way we also have a more brief introduction section.

File: dev/reference/performance/statement-summary.md (hunk 9 above)
Comment (djshow832) [reply dev/reference/performance/statement-summary.md]: Addressed.

File: dev/reference/performance/statement-summary.md (hunk 8 above)
Comment (djshow832) [reply dev/reference/performance/statement-summary.md]: Addressed.

File: dev/reference/performance/statement-summary.md (hunk 7 above)
Comment (djshow832) [reply dev/reference/performance/statement-summary.md]: Addressed.

File: dev/reference/performance/statement-summary.md (hunk 6 above)
Comment (djshow832) [reply dev/reference/performance/statement-summary.md]: Addressed.

File: dev/reference/performance/statement-summary.md (hunk 4 above)
Comment (djshow832) [reply dev/reference/performance/statement-summary.md]: Addressed.

File: dev/reference/mysql-compatibility.md [hunk 10]
```diff
@@ -69,6 +69,8 @@ TiDB 实现自增 ID 的原理是每个 tidb-server 实例缓存一段 ID 值用
 
 Performance schema 表在 TiDB 中返回结果为空。TiDB 使用 [Prometheus 和 Grafana](/dev/how-to/monitor/monitor-a-cluster.md) 来监测性能指标。
 
+从 TiDB 3.0.4 版本开始，TiDB 支持表 `events_statements_summary_by_digest`，参见 [Statement Summary Table](/dev/reference/performance/statement-summary.md)。
```
[the remark this answers is not in the document]
Comment (djshow832) [reply dev/reference/mysql-compatibility.md]: Addressed.

File: dev/reference/performance/statement-summary.md [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+
+针对 SQL 性能相关的问题，MySQL 在 `performance_schema` 提供了 [statement summary tables](https://dev.mysql.com/doc/refman/5.6/en/statement-summary-tables.html)，用来监控和统计 SQL。例如其中的一张表 `events_statements_summary_by_digest`，提供了丰富的字段，包括延迟、执行次数、扫描行数、全表扫描次数等，有助于用户定位 SQL 问题。
+
+为此，从 3.0.4 版本开始，TiDB 也提供系统表 `events_statements_summary_by_digest`。本文将详细介绍 `events_statements_summary_by_digest`，以及如何利用它来排查 SQL 性能问题。
+
+## events_statements_summary_by_digest 介绍
+
+`events_statement_summary_by_digest` 是 `performance_schema` 里的一张系统表。顾名思义，它把 SQL 按 digest 分组，统计每一组的 SQL 信息。
+
+digest 是什么呢？它与 slow log 里的 digest 一样，是把 SQL 规范化后算出的唯一标识符。
```
[the remark this answers is not in the document]
Comment (djshow832) [reply dev/reference/performance/statement-summary.md]: Addressed.

File: dev/reference/performance/statement-summary.md [hunk 12]
```diff
@@ -0,0 +1,150 @@
+---
+title: Statement Summary Table
+category: reference
+---
+
+# Statement Summary Table
+
+## 简介
```
Comment (dcalvin) [inline dev/reference/performance/statement-summary.md]: On second thought, we don't need this title. Please remove it. Thanks.

Comment (dcalvin) [inline dev/reference/performance/statement-summary.md]:
```suggestion
```

Comment (djshow832) [reply dev/reference/performance/statement-summary.md]: Addressed.

File: dev/reference/performance/statement-summary.md [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+- 通过 `max-stmt-count` 更改保存的 SQL 种类数量，默认 100 条。当 SQL 种类超过 `max-stmt-count` 时，会移除最近没有使用的 SQL。
+- 通过 `max-sql-length` 更改 `DIGEST_TEXT` 和 `QUERY_SAMPLE_TEXT` 的最大显示长度，默认是 4096。
+
+这两个参数建议根据实际情况调整，不宜设置得过大。
+
+## 目前的限制
+
+`events_statements_summary_by_digest` 现在还存在一些限制：
+
+- 查询 `events_statements_summary_by_digest` 时，只会显示当前 TiDB-Server 的 statement summary，而不是整个群集的 statement summary。
```
Comment (dcalvin) [inline dev/reference/performance/statement-summary.md]:
```suggestion
- 查询 `events_statements_summary_by_digest` 时，只会显示当前 TiDB-Server 的 statement summary，而不是整个集群的 statement summary。
```

**Comment by lilin90** [comment]
@djshow832 Please add a reason why you closed this PR and add the related PR if possible. @dcalvin FYI.

**Comment by lilin90** [comment]
This PR is closed. Updates are made in a new PR https://github.com/pingcap/docs-cn/pull/1939.

**Pull request closed without merging**